### DISCUSSION TITLE
The logs are never pushed

### DISCUSSION THREAD

**Discussion by camillebourget** [serilog-contrib/serilog-sinks-grafana-loki#133] [Q&A]
Hello,

I tested your product on a .Net Core 6 project which works perfectly and I can push my logs correctly. 

However, when I wanted to copy the same code to a .Net Framework 4.8 class library, I don't receive any logs.

To push my logs I implement a custom HTTP client which adds a header with a token in parameter which allows to authorize the writing of the logs. By debugging my code I noticed that it did not pass any more by my custom implementation and thus did not transmit any more the token which it allowed to write logs.

Here is my custom implementation:

```csharp
WriteTo.GrafanaLoki("https://logs.prd.obs.fr-par.scw.cloud",
                new List<LokiLabel>
                {
                    new()
                    {
                        Key = "********",
                        Value = "*********"
                    }
                },
                httpClient: new CustomHttpClient()
```
```csharp
public class CustomHttpClient : BaseLokiHttpClient
    {
        public override async Task<HttpResponseMessage> PostAsync(string requestUri, Stream contentStream)
        {
            using var content = new StreamContent(contentStream);
            content.Headers.ContentType = MediaTypeHeaderValue.Parse("application/json");
            content.Headers.Add("X-Token", "***********************");
            return await HttpClient.PostAsync(requestUri, content).ConfigureAwait(false);
        }
    }
```

did I forget something ?

I would like to specify that I have tested with all versions of the frameork

thank you for your help
Camille

**Comment by mishamyte** [comment]
Hi @camillebourget,

Is your custom client works with this sink on .NET 6 project?
Have you tried to enable SelfLog (as described in the doc) and check if there is any information?

**Reply by camillebourget** [reply]
yes my .net6 client is working correctly and I have no errors in the SelfLog

I'm still looking and I think it's a library version problem

**Reply by mishamyte** [reply]
Uhm, interesting. To be honest, a little guys used for a classic .NET FW.

Will be grateful if you would have a possibility to create a minimal repro for investigation of your problem

**Reply by camillebourget** [reply]
I am isolating my code to identify the source of the problem because after several tests and from what you tell me it seems that the bug does not come from serilog-sinks-grafana-loki.

I continue my research in any case thank you for your responsiveness I hold you to the cournat if I have news

**Reply by mishamyte** [reply]
OK, please keep me updated if possible
